src/rust/engine/process_execution/src/lib.rs

```diff
@@ -485,6 +486,29 @@ pub struct ProcessExecutionEnvironment {
     pub strategy: ProcessExecutionStrategy,
 }
 
+#[derive(DeepSizeOf, Debug, Clone, Hash, PartialEq, Eq, Serialize)]
+pub enum ProcessConcurrency {
+    /// A range of acceptable cpu cores with optional bounds
+    Range {
+        // Minimum number of cores to use, defaults to 1
+        min: Option<usize>,
+        // Maximum number of cores to use, defaults to min
+        max: Option<usize>,
+    },
+    /// Exclusive access to all cores
+    Exclusive,
+}
+
+// TODO Unclear why I need this or when its invoked, compile error for CommandSpec
```

Comment:
Would be good to figure this out...

Reply:
I followed this a bit, its used by `process_executor` when parsing cli arguments

https://github.com/pantsbuild/pants/blob/efab3e40cedce5c31dcab729fe3ffb8b4314be3e/src/rust/engine/process_executor/src/main.rs#L152-L159

But its not clear to me _when_ this is required. It is for remote/docker execution, something else? 

I also had trouble finding where the arguments are serialized to strings, so I don't have tests for that atm.